Lib/fontTools/merge.py

```diff
@@ -919,21 +988,63 @@ def __init__(self, options=None):
 		self.options = options
 
 	def merge(self, fontfiles):
-
-		mega = ttLib.TTFont()
+		# Take first input file sfntVersion
+		sfntVersion = ttLib.TTFont(fontfiles[0]).sfntVersion
+		mega = ttLib.TTFont(sfntVersion=sfntVersion)
 
 		#
 		# Settle on a mega glyph order.
 		#
 		fonts = [ttLib.TTFont(fontfile) for fontfile in fontfiles]
+
+		
+		# Check if all fonts have CFF
+		allOTFs = all(cff is not None for cff in [font.get('CFF ') for font in fonts])
+
+		cffTables = []
+		if allOTFs:
+			for font in fonts:
+				# Desubroutinize with Subsetter (like compreffor does)
+				from io import BytesIO
+
+				stream = BytesIO()
+				font.save(stream, reorderTables=False)
+				stream.flush()
+				stream.seek(0)
+				tmpfont = ttLib.TTFont(stream)
+
+				options = subset.Options()
+				options.desubroutinize = True
+				options.notdef_outline = True
+				subsetter = subset.Subsetter(options=options)
+				subsetter.populate(glyphs=tmpfont.getGlyphOrder())
+				subsetter.subset(tmpfont)
+				data = tmpfont['CFF '].compile(tmpfont)
+				table = ttLib.newTable('CFF ')
+				table.decompile(data, font)
+				font['CFF '] = table
+				cffTables.append(font.get('CFF '))
```

Comment:
you can unconditionally do `font["CFF "]` instead of `font.get("CFF ")` if you are certain, like in this particular case, that font has "CFF " table inside.
well, actually here you can just do `cffTables.append(table)`

Reply:
makes sense